src/NodeosActionReader.ts

```diff
@@ -25,26 +31,48 @@ export class NodeosActionReader extends AbstractActionReader {
    * Returns a promise for the head block number.
    */
   public async getHeadBlockNumber(): Promise<number> {
-    const blockInfo = await this.httpRequest("get", {
-      url: `${this.nodeosEndpoint}/v1/chain/get_info`,
-      json: true,
-    })
-    if (this.onlyIrreversible) {
-      return blockInfo.last_irreversible_block_num
+    let numTries = 1
+    while (numTries < 6) {
```

Comment:
This is inherently arbitrary, but I think we should support up to 30 seconds of retries, which would be  `(numTries <= 120)` at 250ms try intervals. It's my opinion that 1.25 seconds is not resilient enough.

Maybe we can have the interval + number of tries as default arguments to the method, so they can be configured by the implementer if desired.